frontend/src/pages/surveys/components/step/step.tsx

```diff
@@ -0,0 +1,258 @@
+import { Button, Checkbox, Input, Radio } from '#libs/components/components.js';
+import { FIRST_INDEX, LAST_INDEX } from '#libs/constants/constants.js';
+import {
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useFormController,
+} from '#libs/hooks/hooks.js';
+import {
+  type HandleFieldChangeType,
+  type StepsType,
+} from '#packages/survey/libs/types/types.js';
+import {
+  stepInputValidationSchema,
+  SurveyValidationRule,
+} from '#packages/survey/survey.js';
+import { TEXTAREA_ROWS_COUNT } from '#pages/surveys/libs/constants/constants.js';
+
+import styles from './styles.module.scss';
+
+type SurveyState = {
+  preferences?: string[];
+  feelings?: string[];
+  goals?: string[];
+  worries?: string[];
+  meditationExperience?: string;
+  journalingExperience?: string;
+  other?: string;
+};
+
+const hasOther = (category: string[]): boolean => {
+  return category.includes('Other');
+};
+
+const getOtherDefault = (categories: string[]): string => {
+  return hasOther(categories) && categories.at(LAST_INDEX) !== 'Other'
+    ? (categories.at(LAST_INDEX) as string)
+    : '';
+};
+
+const getOthersCategories = (
+  categories: string[],
+  payload: string[],
+): string[] => {
+  return payload.filter((category) => {
+    return !categories.includes(category);
+  });
+};
+
+const onFieldChange = ({
```

Comment:
remove this, either extend component or create a custom hook

Reply:
done